File: skills/automating-reminders/scripts/create_reminder.js
#!/usr/bin/env osascript -l JavaScript
/**
 * create_reminder.js - Create a reminder in Apple Reminders via JXA
 *
 * Usage:
 *   osascript -l JavaScript create_reminder.js
 *
 * This script creates a reminder with a timed alert. It uses a fallback
 * approach to find an available list since list names vary by system
 * (e.g., "Reminders", "Inbox", or localized names).
 */

function getList(app, preferredName) {
  // Try to get list by preferred name first
  try {
    const list = app.lists.byName(preferredName);
    list.name(); // Verify it exists
    return list;
  } catch (e) {
    // Fall back to first available list
    const lists = app.lists();
    if (lists.length === 0) {
      throw new Error("No reminder lists found");
    }
    return lists[0];
  }
}

function createReminder(options) {
  const app = Application("Reminders");
  const defaults = {
    listName: "Reminders",
    name: "New Reminder",
    body: "",
    minutesFromNow: 5,
    priority: 0 // 0=none, 1=high, 5=medium, 9=low
  };

  const opts = Object.assign({}, defaults, options);

  try {
    const list = getList(app, opts.listName);

    // Calculate reminder time
    const reminderTime = new Date(Date.now() + opts.minutesFromNow * 60 * 1000);

    const reminderProps = {
      name: opts.name,
      remindMeDate: reminderTime
    };

    if (opts.body) {
      reminderProps.body = opts.body;
    }

    if (opts.priority > 0) {
      reminderProps.priority = opts.priority;
    }

    if (opts.dueDate) {
      reminderProps.dueDate = opts.dueDate;
    }

    const reminder = app.Reminder(reminderProps);
    list.reminders.push(reminder);

    return {
      success: true,
      list: list.name(),
      reminderTime: reminderTime.toLocaleString(),
      name: opts.name
    };
  } catch (error) {
    return {
      success: false,
      error: error.message
    };
  }
}

// Example usage - create a reminder for 5 minutes from now
function run() {
  const result = createReminder({
    name: "Call back 555-555-5555",
    body: "Return phone call",
    minutesFromNow: 5,
    priority: 1 // High priority
  });

  if (result.success) {
    return `Reminder "${result.name}" created in '${result.list}' for ${result.reminderTime}`;
  } else {
    return `Error: ${result.error}`;
  }
}

File: skills/automating-reminders/scripts/create_reminder.py
#!/usr/bin/env python3
"""
Create Reminder Script - PyXA Implementation
Creates a new reminder in Reminders app

Usage: python create_reminder.py "Reminder Title" ["due date"] ["list name"]
"""

import sys
from datetime import datetime, timedelta
import PyXA

def create_reminder(title, due_date_str=None, list_name="Reminders"):
    """Create a new reminder"""
    try:
        reminders = PyXA.Application("Reminders")

        # Parse due date if provided
        due_date = None
        if due_date_str:
            try:
                due_date = datetime.fromisoformat(due_date_str)
            except:
                print(f"Invalid date format: {due_date_str}. Use YYYY-MM-DD or YYYY-MM-DD HH:MM")
                return False

        # Find or create list
        target_list = None
        lists = reminders.lists()

        for lst in lists:
            if lst.name == list_name:
                target_list = lst
                break

        if not target_list:
            # Create new list
            target_list = reminders.new_list(list_name)

        # Create reminder
        reminder_data = {"name": title}
        if due_date:
            reminder_data["due_date"] = due_date

        reminder = target_list.reminders().push(reminder_data)

        print(f"Created reminder: '{title}'")
        if due_date:
            print(f"Due: {due_date.strftime('%Y-%m-%d %H:%M')}")
        print(f"List: {list_name}")

        return True

    except Exception as e:
        print(f"Error creating reminder: {e}")
        return False

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python create_reminder.py 'Reminder Title' ['YYYY-MM-DD'] ['List Name']")
        sys.exit(1)

    title = sys.argv[1]
    due_date = sys.argv[2] if len(sys.argv) > 2 else None
    list_name = sys.argv[3] if len(sys.argv) > 3 else "Reminders"

    success = create_reminder(title, due_date, list_name)
    sys.exit(0 if success else 1)